### ISSUE TITLE
Errors in cloud runners are not reliably being captured as failures.

### ISSUE THREAD

**Issue by grichmond-salt** [saltstack/salt#38816]
### Description of Issue/Question
I am working on an orchestration that will get an IP from an IPAM system, initiate vm creation in vmware, and if the vm creation fails, the IP address is released.

However, if there is a problem during the creation of the VM, the Error is being reported under "Changes" and the "Result" is incorrectly reported as "True".  This is breaking the "onfail" requisite.

### Output
```
         ID: provision_vm
    Function: salt.runner
        Name: cloud.create
      Result: True
     Comment: Runner function 'cloud.create' executed.
     Started: 07:38:08.678788
    Duration: 300.053 ms
     Changes:
              Exception occurred in runner cloud.create: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/client/mixins.py", line 395, in _low
                  data['return'] = self.functions[fun](*args, **kwargs)
              TypeError: create() takes at least 2 arguments (0 given)
----------
          ID: release_ip
    Function: salt.function
        Name: cmd.run
      Result: True
     Comment: State was not run because onfail req did not change
     Changes:
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.1

Dependency Versions:
           cffi: 1.6.0
       cherrypy: 3.2.2
       dateutil: 1.5
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Aug  2 2016, 04:20:16)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: redhat 7.3 Maipo
        machine: x86_64
        release: 3.10.0-123.el7.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 7.3 Maipo
```

**Comment by Ch3LL** [comment]
Looks like we need to improve the error detection here. Thanks

**Comment by Ch3LL** [comment]
@grichmond-salt follow up questions: is this also the case for other runners or just the cloud runner?

**Comment by grichmond-salt** [comment]
@Ch3LL I expect that @amendlik would suggest that this should be true for all runners, with the cloud runner taking the top priority.

**Comment by Ch3LL** [comment]
K thanks for clarifying. :)

**Comment by thatch45** [comment]
Ok, the thing to do here is to change the `salt.runner` state to change the result to False if the underlying runner stack traces like it has in this case

**Comment by amendlik** [comment]
The result needs to indicate all types of failures, not just stack traces. It's not clear to me whether the `cloud.profile` runner fails to provide a useful return code, if `state.runner` is ignoring it, or both.

Also, I would request that this be fixed in the next release, 2016.11.2. @thatch45 and I have discussed this.

**Comment by thatch45** [comment]
For the record, I don't like making behavior changes like this in point releases. I think we can make an exception this time, but in general this is not something I would normally accept for a point release

**Comment by grichmond-salt** [comment]
@amendlik unfortunately 2016.11.2 is already tagged and packaged.  However, this fix will definitely be in 2016.11.3.

The update consists of one line being replaced by three lines in salt/states/saltmod.py.  It would be _extremely_ valuable if you could manually patch the current test environment and confirm that the behavior works as expected.
